libs/dal/src/repositories/job/job.repository.ts

```diff
@@ -127,4 +131,35 @@ export class JobRepository extends BaseRepository<EnforceEnvironmentQuery, JobEn
 
     return result;
   }
+
+  public async findOnePopulate({
+    query,
+    select = '',
+    selectTemplate = '',
+    selectNotification = '',
+    selectSubscriber = '',
+    selectEnvironment = '',
+  }: {
+    query: { _environmentId: string; transactionId: string };
+    select?: ProjectionType<JobEntity>;
+    selectTemplate?: ProjectionType<NotificationTemplateEntity>;
+    selectNotification?: ProjectionType<NotificationEntity>;
+    selectSubscriber?: ProjectionType<SubscriberEntity>;
+    selectEnvironment?: ProjectionType<EnvironmentEntity>;
+  }): Promise<
+    JobEntity & {
+      template: NotificationTemplateEntity;
+      notification: NotificationEntity;
+      subscriber: SubscriberEntity;
+      environment: EnvironmentEntity;
+    }
+  > {
+    return Job.findOne(query, select)
+      .populate('template', selectTemplate)
+      .populate('notification', selectNotification)
+      .populate('subscriber', selectSubscriber)
+      .populate('environment', selectEnvironment)
+      .lean()
+      .exec();
```

Comment:
I think we can make actual calls to the repository ourself and not rely on mongoose for those. So we can cache them independently. And I think we can skip notification or template selection, as long as we return the user the identifier of this notification template and it's _id